pgml_rust/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+use pgx::*;
+use xgboost::{parameters, DMatrix, Booster};
+use std::path::Path;
+
+pg_module_magic!();
+
+
+/// Main training function to train an XGBoost model on a dataset.
+/// Parameters:
+///   - table: name of the table/view
+///   - features: array of column names to use as features
+///   - label: the name of the target column
+///
+/// Example:
+///
+/// ```
+/// SELECT * FROM pgml_train('pgml.diabetes', ARRAY['age', 'sex'], 'target');
+#[pg_extern]
```

Comment:
The `pgml` schema is not used in this one, at least not yet. I'm building this parallel to `pgml` in `pgml_rust`, so this can be used as an optimization for available algorithms.